app/ts/components/Transfer.tsx

```diff
@@ -1,39 +1,138 @@
-import { Signal, useSignal } from "@preact/signals";
+import { Signal, useComputed, useSignal, useSignalEffect } from "@preact/signals";
+import { getAddress } from "ethers";
+import { JSX } from "preact/jsx-runtime";
 import { connectBrowserProvider, ProviderStore } from "../library/provider.js";
+import { detectNft } from "../library/identifyTokens.js"
 import { BlockInfo } from "../library/types.js";
 import { Button } from "./Button.js";
 import { TokenPicker } from "./TokenPicker.js";
+import { transferNft } from "../library/transactions.js";
 
 export const Transfer = ({ provider, blockInfo }: { provider: Signal<ProviderStore | undefined>, blockInfo: Signal<BlockInfo> }) => {
 	const showTokenPicker = useSignal<boolean>(false)
+	const selectedNft = useSignal<{ address: string, id: bigint, owner: string, name: string | undefined, tokenURI: string | undefined } | undefined>(undefined)
 
-	function transferNft() { }
+	const fetchingStates = useSignal<'empty' | 'fetching' | 'notfound' | 'badid' | 'noprovider'>('empty')
+	const sendText = useComputed(() => {
+		if (!selectedNft.value) return 'Input Token Details'
+		if (selectedNft.value.owner !== provider.value?.walletAddress) return 'You do not own this token'
+		if (!recipientInput.value) return 'Missing Recipient'
+		if (recipientInput.value === provider.value.walletAddress) return 'Cannot send to yourself'
+		if (recipientInput.value === selectedNft.value.address) return 'Cannot send to NFT contract'
+		return 'Send'
+	})
 
+	const addressInput = useSignal<string | undefined>(undefined)
+	const idInput = useSignal<bigint | undefined>(undefined)
+	const recipientInput = useSignal<string | undefined>(undefined)
+
+	useSignalEffect(() => {
+		attemptToFetchNft(addressInput.value, idInput.value)
+	})
+
+	function validateAddressInput(event: JSX.TargetedEvent<HTMLInputElement>) {
+		try {
+			addressInput.value = getAddress(event.currentTarget.value.toLowerCase())
+		} catch (e) {
+			console.error(e)
+			addressInput.value = undefined
+		}
+	}
+
+	function validateRecipientInput(event: JSX.TargetedEvent<HTMLInputElement>) {
+		try {
+			recipientInput.value = getAddress(event.currentTarget.value.toLowerCase())
+		} catch (e) {
```

Comment:
we shouldn't be using this try {} catch pattern for errors (many places in this pr for these), I believe should see what these errors can be and handle without try catch. And then show them to the user that something went wrong.